MathlibTest/abel.lean

```diff
@@ -143,21 +143,18 @@ example [AddCommGroup α] (x y z : α) (h : False) (w : x - x = y + z) : False :
   assumption
 
 abbrev myId (a : ℤ) : ℤ := a
+irreducible_def myIdOpaque : ℤ → ℤ := myId
 
 /-
 Test that when `abel_nf` normalizes multiple expressions which contain a particular atom, it uses a
 form for that atom which is consistent between expression.
-
-Note: This test is useless unless done at `=ₛ` (the level of syntactic equality); doing this test at
-`=` (the level of reducible equality) would not catch the erroneous old behaviour (normalizing one
-expression to `2 • myId x` and the other to `2 • x`).  But to get syntactic equality, we need to
-make the typeclass arguments agree, which requires some handholding in typeclass inference ... which
-is the reason for all the locally deleted instances here.
 -/
-attribute [-instance] Int.instAddGroup AddGroupWithOne.toAddGroup in
 example (x : ℤ) (R : ℤ → ℤ → Prop) (hR : Reflexive R) : True := by
   have : R (myId x + x) (x + myId x) := by
     abel_nf
-    guard_target =ₛ R ((2:ℤ) • myId x) ((2:ℤ) • myId x)
+    -- `guard_target` is using reducible defeq, so we need to make sure it cannot unfold any `myId`s
+    -- in the goal state.
+    rw [← myIdOpaque_def]
+    guard_target = R ((2:ℤ) • myIdOpaque x) ((2:ℤ) • myIdOpaque x)
```

Comment:
@hrmacbeth, feel free to change this comment if you don't think it captures the intent well